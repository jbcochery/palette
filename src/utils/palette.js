export function generatePalette(start, end, steps) {
  const palette = [];
  for (let x = 0; x < steps; x++) {
    const row = [];
    for (let y = 0; y < steps; y++) {
      row.push("#000000");
    }
    palette.push(row);
  }
  return palette;
}
