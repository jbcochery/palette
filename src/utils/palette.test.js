import { generatePalette } from "./palette";

it("generates a palette filled with #000000", () => {
  expect(generatePalette("#ff0000", "#ffff00", 4)).toMatchSnapshot();
});
