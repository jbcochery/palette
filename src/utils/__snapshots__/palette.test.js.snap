// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates a palette filled with #000000 1`] = `
Array [
  Array [
    "#000000",
    "#000000",
    "#000000",
    "#000000",
  ],
  Array [
    "#000000",
    "#000000",
    "#000000",
    "#000000",
  ],
  Array [
    "#000000",
    "#000000",
    "#000000",
    "#000000",
  ],
  Array [
    "#000000",
    "#000000",
    "#000000",
    "#000000",
  ],
]
`;
